### PR TITLE
feat: add new scanner based on GitHub commits (GithubScanner)

### PR DESCRIPTION
**Pull request by javierron**
Refactoring  the GitHub scanner and pipeline integration.

### CODE REVIEW

**Comment by monperrus** [comment]
Nice! What remains to be done to unwip?

File: src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/dockerpool/DockerPoolManager.java [hunk 1]
```diff
@@ -62,17 +64,30 @@ public void setEngines(List<SerializerEngine> engines) {
      * @param buildId
      * @return
      */
-    public TreatedBuildTracking prepareBeforeSubmitBuild(long buildId) {
```
Comment (monperrus) [inline src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/dockerpool/DockerPoolManager.java:65]: This class should remain independent of the builds

**Review by monperrus — COMMENTED** [review]
LGTM, thanks!

My only concern is in DockerPool see comment.

**Comment by monperrus** [comment]
Plus we need a new test for GithubScanner

File: src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/dockerpool/DockerPoolManager.java (hunk 1 above)
Comment (javierron) [reply src/repairnator-core/src/main/java/fr/inria/spirals/repairnator/dockerpool/DockerPoolManager.java:65]: Done! This change led me to abstract out some JTravis references too, I think that's a good thing if we are to phase out Travis code.

**Comment by monperrus** [comment]
Great work, thanks @javierron 

This will be very useful for the Sorald experiments lead by @khaes-kth

**Comment by khaes-kth** [comment]
Thanks a lot @javierron

**Pull request merged**